torchvision/csrc/io/decoder/stream.cpp

```diff
@@ -49,7 +51,51 @@ int Stream::openCodec(std::vector<DecoderMetadata>* metadata) {
                << ", avcodec_alloc_context3 failed";
     return AVERROR(ENOMEM);
   }
+  // multithreading heuristics
+  // this is a dumb conservative limit; ideally we'd use 
+  // int max_threads = at::get_num_threads(); but this would cause 
+  // fb sync to fail as it would add dependency to ATen to the decoder API
+  int max_threads = 12;
+  if (num_threads > max_threads) {
+    num_threads = max_threads;
+  }
+
+  if (num_threads > 0) {
+    if (num_threads > 1) {
+      codecCtx_->active_thread_type = 1;
+    }
+    // if user defined, respect that
+    codecCtx_->thread_count = num_threads;
+
+  } else {
+    // otherwise set sensible defaults
+    // with the special case for the different MPEG4 codecs
+    // that don't have threading context functions
+    // TODO: potentially automate this using native c++ function lookups
+    if (strcmp(codecCtx_->codec->name, "mpeg4") == 0 &&
+        codecCtx_->codec_type == 0) {
+      if (codecCtx_->codec_tag == 1684633208) {
```

Comment:
What is this magic number here?

Reply:
This is codec signature for videos from HMDB and some from UCF.
In principle, we can avoid this by checking if the codec implements threading context function, but I'm  not sure how to do it; I'll ask around.

Reply:
Okay sure, the alternative solution sounds better.